### PR TITLE
Force wp-api.js to use HTTP/1.0

### PR DESCRIPTION
**Pull request by pento**
## Description

We've had several problems in the past where HTTP/1.1 methods are blocked, due to server oddities, or firewall rules. In particular, #2704 is an ongoing issue where Cloudflare firewall rules block `PUT` requests, to mitigate a security issue that existed in WordPress 4.7 and 4.7.1.

Switching to `POST` requests may help work around this issue.

### CODE REVIEW

**Review by atimmer — APPROVED** [review]
We've had the same issues when we included a PUT request in our plugin. It is too bad, but it seems to be a reality that we cannot safely use PUT, PATCH and DELETE methods.

I like `emulateHTTP` because it is still handled as a PUT request in the REST API code. 👍🏻

**Comment by aduth** [comment]
Are there any downsides to this? If not, is there any reason it shouldn't be the default, not Gutenberg-specific?

**Comment by pento** [comment]
I haven't been able to find any previous discussion on it, I assume it was just never discussed. 

The only downside I'm aware of is the slight increase in payload size to include the `X-Http-Method-Override` header and `_method` parameter. The REST API treats them the same, and it should have better compatibility with old or obscure proxies and servers that don't implement HTTP/1.1.

**Comment by aduth** [comment]
I had a sense that I'd recalled being involved in a previous discussion here:

https://wordpress.slack.com/archives/C02RQC26G/p1509540912000307
https://wordpress.slack.com/archives/C02RQC26G/p1509542116000023

tl;dr:

>Caching is one potential issue, as are status codes
>Some methods can be cached (like `DELETE`), but `POST` can't be
>And status codes like 201 may change meaning for POST vs PUT
>But in practice, unlikely to cause a huge issue

**Comment by pento** [comment]
Okay, I'm inclined to be more concerned with compatibility than caching - for folks where caching might be useful, they can turn it off.

**Comment by aduth** [comment]
Will the intended effect of this have regressed with #5253 , where we're using `wp.apiRequest` instead of `wp.api` ? cc @youknowriad

**Comment by aduth** [comment]
Seems like this would be a good option for `wp.apiRequest` (to set defined method as header).

**Comment by pento** [comment]
Yeah, this definitely regressed with #5253. There are a bunch of new bug reports that are likely caused by overly aggressive firewall rules for `PUT` requests. Eg, #5675, #5632, #5660.
